### ISSUE TITLE
Improve performance for peer search when finding/allocating peer

### ISSUE THREAD

**Issue by mwyzhao** [Multi-Tier-Cloud/service-manager#55]
To be done after #3. Currently, it pings and and sorts all peers to find the best peer. This is really bad performance wise but can be improved by using Prometheus, and stopping after a good enough peer is found.

**Comment by mwyzhao** [comment]
Duplicate of common#3
